test/compositedataframes.jl

```diff
@@ -55,4 +57,26 @@ for x in eachrow(df)
 end
 @test res == sum(df.A .* df.B)
 
+itr = eachrow(df)
+@test size(itr) == (3,)
+@test length(itr) == 3
+@test getindex(itr, 1) == row(itr.df, 1)
+@test map(itr) do row
+    row.A * row.B
+end == [2, 2, 6]
+
+df2 = DataFrame(C = [1, 2, 3], D = [2, 1, 2])
+df3 = CompositeDataFrame(:DF3, C = [1, 2, 3], D = [2, 1, 2])
+
+@test names(hcat(df, df2)) == [:A, :B, :C, :D]
+@test names(hcat(df2, df)) == [:C, :D, :A, :B]
+@test names(hcat(df, df3)) == [:A, :B, :C, :D]
+
+@test CompositeDataFrame(DataFrame(df)) == df
+@test names(CompositeDataFrame(DataFrame(df), [:C, :D])) == [:C, :D]
+@test row() == nothing
+@test df[1, :A] == 1
+@test names( df[[1, 2]] ) == [:A, :B]
```

Comment:
I was trying hit coverage of the `Base.getindex{T <: DataFrames.ColumnIndex}(cdf::AbstractCompositeDataFrame, col_inds::AbstractVector{T})` constructor